keras_cv/models/convnext.py

```diff
@@ -0,0 +1,494 @@
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+      model_name: An optional name for the model.
+
+    Returns:
+      A `keras.Model` instance.
+
+    Raises:
+        ValueError: in case of invalid argument for `weights`,
+          or invalid input shape.
+        ValueError: if `classifier_activation` is not `softmax`, or `None`
+          when using a pretrained top layer.
+        ValueError: if `include_top` is True but `num_classes` is not 1000
+          when using ImageNet.
+    """
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` or the path to the weights file to be loaded. "
+            f"Weights file not found at location: {weights}"
+        )
+
+    if include_top and not classes:
+        raise ValueError(
+            "If `include_top` is True, "
+            "you should specify `classes`. "
+            f"Received: classes={classes}"
+        )
+
+    if include_top and pooling:
+        raise ValueError(
+            f"`pooling` must be `None` when `include_top=True`."
+            f"Received pooling={pooling} and include_top={include_top}. "
+        )
+
+    inputs = utils.parse_model_inputs(input_shape, input_tensor)
+
+    x = inputs
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
+
+    # Stem block.
+    stem = keras.Sequential(
+        [
+            layers.Conv2D(
+                projection_dims[0],
+                kernel_size=4,
+                strides=4,
+                name=model_name + "_stem_conv",
+            ),
+            layers.LayerNormalization(
+                epsilon=1e-6, name=model_name + "_stem_layernorm"
+            ),
+        ],
+        name=model_name + "_stem",
+    )
+
+    # Downsampling blocks.
+    downsample_layers = []
+    downsample_layers.append(stem)
+
+    num_downsample_layers = 3
+    for i in range(num_downsample_layers):
+        downsample_layer = keras.Sequential(
+            [
+                layers.LayerNormalization(
+                    epsilon=1e-6,
+                    name=model_name + "_downsampling_layernorm_" + str(i),
+                ),
+                layers.Conv2D(
+                    projection_dims[i + 1],
+                    kernel_size=2,
+                    strides=2,
+                    name=model_name + "_downsampling_conv_" + str(i),
+                ),
+            ],
+            name=model_name + "_downsampling_block_" + str(i),
+        )
+        downsample_layers.append(downsample_layer)
+
+    # Stochastic depth schedule.
+    # This is referred from the original ConvNeXt codebase:
+    # https://github.com/facebookresearch/ConvNeXt/blob/main/models/convnext.py#L86
+    depth_drop_rates = [float(x) for x in np.linspace(0.0, drop_path_rate, sum(depths))]
```

Comment:
Can we remove the numpy dependency in non-testing code with native TF ops?

https://github.com/tensorflow/tensorflow/blob/v2.9.1/tensorflow/python/ops/numpy_ops/np_math_ops.py#L1049-L1090

Reply:
I remember using the native TF ops failed in an internal test when I contributed the same model to `keras.applications`. 

Cc: @LukeWood

Reply:
Could you please demonstrate that our tests fail if using a native TF op here?
I agree that we should prefer a native TF op.

Reply:
Addressed it.